api/rest/client_interface/client_interface.go

```diff
@@ -0,0 +1,43 @@
+package client_interface
+
+import (
+	"context"
+
+	"github.com/ipfs/ipfs-cluster/api"
+
+	cid "github.com/ipfs/go-cid"
+	shell "github.com/ipfs/go-ipfs-api"
+	files "github.com/ipfs/go-ipfs-cmdkit/files"
+	peer "github.com/libp2p/go-libp2p-peer"
+)
+
+type ClientIface interface {
```

Comment:
If we move this file to the `rest` package, I would rename this to just `Client`. This will be inline with the naming of our other interfaces, for examples see the ipfscluster file: https://github.com/ipfs/ipfs-cluster/blob/753322cdc1e6d5459bace4fb2fd647f529a56278/ipfscluster.go#L70-L94